core/src/client/test_client.rs

```diff
@@ -438,7 +438,7 @@ impl ImportBlock for TestBlockChainClient {
         }
         let len = self.numbers.read().len();
         if number == len {
-            mem::replace(&mut *self.last_hash.write(), h);
```

Comment:
How about using `swap`?

Reply:
https://doc.rust-lang.org/std/mem/fn.swap.html

Reply:
If we use swap here, the following lines are affected from the content swap.
This function returns the value h as `Ok(h)`, `swap` will produce unwanted changes in code.

Reply:
I think it's okay if we call swap on the cloned `h`, but I'm not sure it's more readable. Let's go ahead with this code.